### PR TITLE
[FILE INPUT] fix multiple fix extension bug

### PR DESCRIPTION
**Pull request by jaredcunha**
## Description

Should fix issue #3848 

## Additional information

Include any of the following (as necessary): 

* Switched file input example to `multiple` to ensure script works when selecting multiple files of different type 
* Tested mix of accepted and not accepted files
Screen Shot 2020-11-24 at 1 53 28 PM

Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [x] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `npm test` and make sure the tests for the files you have changed have passed.
- [x] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.
- [x] Title your pull request using this format: [Website] - [UI component]: Brief statement describing what this pull request solves.

### CODE REVIEW

**Review by ConnorDY — APPROVED** [review]
This looks great to me and solves the issues I described.

However, I think a separate issue should be created for what was brought up here:
https://github.com/uswds/uswds/issues/3848#issuecomment-733091993

File: src/js/components/file-input.js [hunk 1]
```diff
@@ -142,27 +142,29 @@ const removeOldPreviews = (dropTarget, instructions) => {
...
     for (let i = 0; i < e.dataTransfer.files.length; i += 1) {
       const file = e.dataTransfer.files[i];
       if (allFilesAllowed) {
-        allFilesAllowed = file.name.indexOf(acceptedFiles);
-        if (allFilesAllowed < 0) {
-          break;
+        for (let j = 0; j < acceptedFiles.length; j += 1) {
+          const fileType = acceptedFiles[j];
+          allFilesAllowed = file.name.indexOf(fileType) > 0 || file.type === fileType;
+          if (allFilesAllowed) break;
```
Comment (ConnorDY) [inline src/js/components/file-input.js:161]: This only breaks out of the file types/extensions loop. Might be worth adding an `else break;` after the the `if (allFilesAllowed) { /* ... */ }` block.

File: src/js/components/file-input.js [hunk 2]
```diff
@@ -142,27 +142,29 @@ const removeOldPreviews = (dropTarget, instructions) => {
...
         }
```
Comment (ConnorDY) [inline src/js/components/file-input.js:162]: There's actually more that needs to be done here. This fails if the first file type isn't the same as the first type listed in the `accept`.

**Review by ConnorDY — CHANGES_REQUESTED** [review]

File: src/js/components/file-input.js (hunk 2 above)
Comment (ConnorDY) [inline src/js/components/file-input.js:162]: There's actually a little more that needs to be done here. This actually doesn't work if the first file isn't the same type as the first file type/extension from the `accept` attribute.

**Review by ConnorDY — CHANGES_REQUESTED** [review]

File: src/js/components/file-input.js (hunk 2 above)
Comment (ConnorDY) [reply src/js/components/file-input.js:162]: I was wrong about this. Please disregard.

**Review by ConnorDY — APPROVED** [review]

File: src/js/components/file-input.js (hunk 2 above)
Comment (jaredcunha) [reply src/js/components/file-input.js:162]: managed to work on my end - attribute is `accept=".pdf,.txt"`
Screen Shot 2020-11-24 at 2 10 57 PM
Screen Shot 2020-11-24 at 2 11 30 PM

**Comment by jaredcunha** [comment]
> However, I think a separate issue should be created for what was brought up here:
#3848 (comment)

Managed to push a fix that I think might support the wildcard

**Review by thisisdano — APPROVED** [review]

**Comment by jaredcunha** [comment]
@thisisdano - checking back in on this saw. Saw this was flagged
Screen Shot 2020-12-11 at 10 03 47 AM

Thoughts about dismissing? Wildcards would only have one `*`. Would just need to update the replace method to `replace(/\*/g, "")`

File: src/js/components/file-input.js (hunk 1 above)
Comment (thisisdano) [reply src/js/components/file-input.js:218]: Done

**Pull request merged**